frontend/public/components/sidebars/resource-sidebar-samples.tsx

```diff
@@ -228,6 +231,48 @@ const defaultSamples = ImmutableMap<GroupVersionKind, Sample[]>()
         targetResource: getTargetResource(ConsoleLinkModel),
       },
     ],
+  )
+  .setIn(
+    [referenceForModel(ConsoleModel)],
+    [
+      {
+        title: 'Add catalog categories',
+        description:
+          'Provides a list of default categories which are shown in the Developer Catalog.',
+        id: 'default',
+        snippet: true,
+        lazyYaml: () => {
+          const mapSubcategory = (subcategory) => ({
+            id: subcategory.id,
+            label: subcategory.label,
+            tags: !subcategory.field || subcategory.field === 'tag' ? subcategory.values : [],
+          });
+
+          const mapCategory = (category) => ({
+            id: category.id,
+            label: category.label,
+            tags: !category.field || category.field === 'tag' ? category.values : [],
+            subcategories: category.subcategories
+              ? Object.values(category.subcategories).map(mapSubcategory)
+              : [],
+          });
+
+          const categories = Object.values(catalogCategories).map(mapCategory);
+
+          const yaml = {
+            spec: {
+              customization: {
+                developerCatalog: {
+                  categories,
+                },
+              },
+            },
+          };
+          return YAML.dump(yaml);
```

Comment:
can make use if `safeJSToYAML`

Reply:
Don't see a reason why our generated structure should fail. But that could be also "famous last words"..

But I don't see a good fallback here, so I would prefer to keep it simple and just dump the data.

I could extract this into a utils function and write a unit test for this to ensure that it doesn't fail in the future. Okay with that?

Reply:
yeah as it's own json so should be good :)